### ISSUE TITLE
Doesn't load with v0.5 master

### ISSUE THREAD

**Issue by ScottPJones** [JuliaGL/GLVisualize.jl#76]
I built v0.5, also did a checkout of GFLW (see https://github.com/JuliaGL/GLFW.jl/pull/78) to fix a deprecation warning, as well as FixedSizeArrays, DualNumbers, Mustache to attempt to fix other warnings, but they are still present.
```julia
19:18 $ julia
               _
   _       _ _(_)_     |  A fresh approach to technical computing
  (_)     | (_) (_)    |  Documentation: http://docs.julialang.org
   _ _   _| |_  __ _   |  Type "?help" for help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 0.5.0-dev+3207 (2016-03-18 22:14 UTC)
 _/ |\__'_|_|_|\__'_|  |  Commit b4637e3* (0 days old master)
|__/                   |  x86_64-apple-darwin15.4.0

julia> using GLVisualize
INFO: Precompiling module GLVisualize...
INFO: Recompiling stale cache file /Users/scott/.julia/lib/v0.5/GLFW.ji for module GLFW.
INFO: Recompiling stale cache file /Users/scott/.julia/lib/v0.5/ModernGL.ji for module ModernGL.
INFO: Recompiling stale cache file /Users/scott/.julia/lib/v0.5/Compat.ji for module Compat.
INFO: Recompiling stale cache file /Users/scott/.julia/lib/v0.5/GLAbstraction.ji for module GLAbstraction.
INFO: Recompiling stale cache file /Users/scott/.julia/lib/v0.5/Quaternions.ji for module Quaternions.
INFO: Recompiling stale cache file /Users/scott/.julia/lib/v0.5/DualNumbers.ji for module DualNumbers.
INFO: Recompiling stale cache file /Users/scott/.julia/lib/v0.5/NaNMath.ji for module NaNMath.
INFO: Recompiling stale cache file /Users/scott/.julia/lib/v0.5/Calculus.ji for module Calculus.
WARNING: Method definition abs2(DualNumbers.Dual4) in module DualNumbers at /Users/scott/.julia/v0.5/DualNumbers/src/dual_n.jl:78 overwritten at /Users/scott/.julia/v0.5/DualNumbers/src/dual_n.jl:154.
INFO: Recompiling stale cache file /Users/scott/.julia/lib/v0.5/FixedSizeArrays.ji for module FixedSizeArrays.
WARNING: Method definition (::Type{#FSA<:FixedSizeArrays.FixedArray})(#X<:Any) in module FixedSizeArrays at /Users/scott/.julia/v0.5/FixedSizeArrays/src/constructors.jl:20 overwritten at /Users/scott/.julia/v0.5/FixedSizeArrays/src/constructors.jl:63.
INFO: Recompiling stale cache file /Users/scott/.julia/lib/v0.5/GeometryTypes.ji for module GeometryTypes.
INFO: Recompiling stale cache file /Users/scott/.julia/lib/v0.5/ColorTypes.ji for module ColorTypes.
INFO: Recompiling stale cache file /Users/scott/.julia/lib/v0.5/FixedPointNumbers.ji for module FixedPointNumbers.
INFO: Recompiling stale cache file /Users/scott/.julia/lib/v0.5/Reactive.ji for module Reactive.
INFO: Recompiling stale cache file /Users/scott/.julia/lib/v0.5/Mustache.ji for module Mustache.
WARNING: Method definition _renderTokensByValue(Function, Any, Any, Any, Any, Any) in module Mustache at /Users/scott/.julia/v0.5/Mustache/src/tokens.jl:220 overwritten at /Users/scott/.julia/v0.5/Mustache/src/tokens.jl:245.
INFO: Recompiling stale cache file /Users/scott/.julia/lib/v0.5/FileIO.ji for module FileIO.
Assertion failed: (addr != 0), function getAddressForOrCompileFunction, file /j/julia/src/codegen.cpp, line 1063.
ERROR: LoadError: Failed to precompile GLWindow to /Users/scott/.julia/lib/v0.5/GLWindow.ji
```

**Comment by SimonDanisch** [comment]
I try to keep GLVisualize functional on 0.5, but haven't had a go at removing warnings yet ;)

**Comment by ScottPJones** [comment]
Right, I wasn't too concerned about the warnings (and I have submitted PRs already to fix both the warnings in GLFW.jl and DualNumbers.jl), but rather that it is not functional, because of the final error due to the assertion failing in codegen.c.

**Comment by SimonDanisch** [comment]
Oh whoops...  This looks a bit like it's rather the fault of 0.5 ?! I'll investigate... Please not that I'm not fully devoted to have `GLVisualize` run on the latest master ;)

**Comment by SimonDanisch** [comment]
I can't reproduce this on:
```Julia
julia> versioninfo()
Julia Version 0.5.0-dev+3177
Commit 2a3aada (2016-03-16 13:12 UTC)
Platform Info:
  System: Linux (x86_64-unknown-linux-gnu)
  CPU: Intel(R) Core(TM) i3-4130 CPU @ 3.40GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Haswell)
  LAPACK: libopenblas64_
  LIBM: libopenlibm
  LLVM: libLLVM-3.7.1 (ORCJIT, haswell)
```
I tested from a clean state just with `Pkg.add("GLVisualize");Pkg.checkout("Mustache")`
The warnings have really gotten out of hand, though :D

**Comment by SimonDanisch** [comment]
I noticed, that sometimes you actually need to remove the cached files from `lib/v0.5`, and then precompile one or two times... Not really sure when this was actually needed ;)

**Comment by ScottPJones** [comment]
Yes, I've been always removing the cached files every time from `lib/v0.5`, and I understand the pain of trying to keep up with the latest master :grinning:, however, part of it is *your* fault for making such a cool package that people want to run it even on the bleeding edge of master! 😜 I'll see if I can eliminate the warnings (2 down and 2 to go) and figure out whatever in v0.5 is breaking it.  Thanks!

**Comment by ScottPJones** [comment]
@SimonDanisch As soon as they get merged, I've dealt with all of the v0.5 warnings (see https://github.com/SimonDanisch/FixedSizeArrays.jl/pull/89)
I'm closing this - I still see this happening, but I've narrowed down to something happening in `GLWindow`, which, as you said, is likely something broken recently in master.
```julia
Julia Version 0.5.0-dev+3211
Commit a013532* (2016-03-20 20:54 UTC)
Platform Info:
  System: Darwin (x86_64-apple-darwin15.4.0)
  CPU: Intel(R) Core(TM) i7-4980HQ CPU @ 2.80GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Haswell)
  LAPACK: libopenblas64_
  LIBM: libopenlibm
  LLVM: libLLVM-3.7.1 (ORCJIT, haswell)
```
